roles/openshift_node/tasks/main.yml

```diff
@@ -20,6 +14,7 @@
       hostname: "{{ openshift_hostname | default(none) }}"
       public_hostname: "{{ openshift_public_hostname | default(none) }}"
       deployment_type: "{{ openshift_deployment_type }}"
+      dns_ip: "{{ openshift_dns_ip if openshift_dns_ip | default(None) else openshift_master_cluster_vip | default(None) }}"
```

Comment:
just nitpicking here, but since you are testing the output of the first default filter, using a boolean value instead of none is a bit more clear.
```
dns_ip: "{{ openshift_dns_ip if openshift_dns_ip | default(false) else openshift_master_cluster_vip | default(None) }}"
```